### PR TITLE
[NIFI-3035] use URLSearchParams .toString() to update URL

### PR DESCRIPTION
**Pull request by scottyaslan**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas-utils.js [hunk 1]
```diff
@@ -412,9 +412,9 @@
 
                 // create object whose keys are the parameter name and the values are the parameter values
                 var paramsObject = {};
-                params.forEach(function (v, k) {
-                    paramsObject[k] = v;
-                });
+                for(var pair of params.entries()) {
```
Comment (gresockj) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas-utils.js]: I think this should be "pair **in** params.entries()"

Comment (gresockj) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas-utils.js]: Btw, after I applied your patch and changed "of" to "in", the issue was indeed resolved and I can now load the console.

Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas-utils.js]: Changing the "of" to "in" is not correct... I tried it and verified that using the "in" breaks deep linking. If you look here (in the Browser Compatibility section): https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams we see that "for...of" is the supported way to iterate over the GET params...

Comment (gresockj) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas-utils.js]: Interesting, I see what you mean.  When I build the nifi-web-ui module with the "of" in place, I get the following error:
[ERROR] /var/local/workspace/nifi/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas-utils.js [1:0]: Compilation produced 301 syntax errors.
[ERROR] ...nf-canvas-utils.js:line 1:column 0:Compilation produced 301 syntax errors.

However, when I change "of " to "in", the error goes away and it is able to build the nar.  How did you get around this issue?

Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas-utils.js]: Good catch! As it turns out we dont need to loop over the URLSearchParams...we can just use the .toString(). I have updated this PR to reflect this change. Please test and let me know if it works for you.

Comment (gresockj) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas-utils.js]: Yes, this works in build and in bringing up the console!  Thanks for the update.

Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-canvas-utils.js]: :)

**Comment by mcgilman** [comment]
Thanks guys. This has been merged to master.

**Pull request closed without merging**